lib/PHPExif/Adapter/AdapterAbstract.php

```diff
@@ -119,14 +121,16 @@ public function setOptions(array $options) : AdapterInterface
     /**
      * Encodes an array of strings into UTF8
      *
-     * @param array|string $data
-     * @return array|string|null
+     * @template T of array|string
+     * @param T $data
+     * @return (T is string ? string : array)
```

Comment:
It shows that I haven't looked over the Lychee PR recently but this is so fancy! :wink: